cdap-sentry/cdap-sentry-extension/cdap-sentry-binding/src/main/java/co/cask/cdap/security/authorization/sentry/binding/AuthBinding.java

```diff
@@ -103,11 +107,111 @@
   private final String instanceName;
   private final String sentryAdminGroup;
 
-  AuthBinding(String sentrySite, String instanceName, String sentryAdminGroup) {
+  private final LoadingCache<Principal, Set<String>> groupCache;
```

Comment:
The group lookup happens on every enforce call. The group mapper implementation can be replaced with a custom implementation which may not have a cache.

Since these new caches will be read-only and won't have any updates to them from the authorizer end (once we remove grants/revokes), I don't see any reason for inconsistency. Also not sure what overhead this cache will add other than a few extra KB of memory used. Let me know if have any other inconsistency or overhead in mind.